File: Dummy/hawkshaw/library/DelegatesExt.java
package com.hawkshaw.library;

public final class DelegatesExt {
    public static final DelegatesExt INSTANCE = new DelegatesExt();

    private DelegatesExt() {
    }

    public final <T> NotNullSingleValueVar<T> notNullSingleValue() {
        return new NotNullSingleValueVar<>();
    }
}

File: app/src/main/java/com/hawkshaw/library/DelegatesExt.kt
package com.hawkshaw.library

/**
 * Extension object providing property delegates
 */
object DelegatesExt {
    /**
     * Creates a property delegate that can only be set once and must be non-null
     * 
     * Usage:
     * ```
     * private val myProperty by DelegatesExt.notNullSingleValue<String>()
     * ```
     */
    fun <T> notNullSingleValue(): NotNullSingleValueVar<T> = NotNullSingleValueVar()
} 